pandas/_config/config.py

```diff
@@ -740,7 +665,10 @@ def _build_option_description(k: str) -> str:
         s += "No description available."
 
     if o:
-        s += f"\n    [default: {o.defval}] [currently: {_get_option(k, True)}]"
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", FutureWarning)
+            warnings.simplefilter("ignore", DeprecationWarning)
+            s += f"\n    [default: {o.defval}] [currently: {get_option(k)}]"
```

Comment:
One thing to be aware of is the impact of https://github.com/python/cpython/issues/73858. Once this block is entered, the cache of previously seen warnings is cleared so that users will see any already encountered warning again. If encountered in a for loop this can cause a warning to get printed thousands of times e.g. #57054.

I believe this will only be run on import, so I think this is likely to have almost no impact - in the odd case you import pandas in e.g. a for loop the warning would be seen at most twice because of import caching.